state/factory/workingset.go

```diff
@@ -183,17 +183,16 @@ func (ws *workingSet) Revert(snapshot int) error {
 
 // Commit persists all changes in RunActions() into the DB
 func (ws *workingSet) Commit(ctx context.Context) error {
-	err := ws.commitFunc(ws.height)
-	if !ws.Dirty() {
+	if err := ws.commitFunc(ws.height); err != nil {
 		return err
 	}
-	if err == nil {
-		protocolCommit(ctx, ws)
-	} else {
-		protocolAbort(ctx, ws)
+	if !ws.dock.Dirty() {
+		if err := protocolCommit(ctx, ws); err != nil {
```

Comment:
also seems in previous logic, ws.Reset is called. but now it doesn't call ws.Reset on error

Reply:
We actually do not need to call "Reset". The data stored in ws.dock should be similar to ws itself, which doesn't need to be explicitly cleaned up.